lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -65,25 +68,34 @@ export abstract class ClientApplication {
      * @param {@link (Configuration:type)} configuration object for the MSAL PublicClientApplication instance
      */
     constructor(configuration: Configuration) {
-        // Set the configuration.
-        this.config = buildConfiguration(configuration);
+        /*
+         * If loaded in an environment where window is not available,
+         * set internal flag to false so that further requests fail.
+         * This is to support server-side rendering environments.
+         */
+        this.isBrowserEnvironment = typeof window !== "undefined";
+
+        if (this.isBrowserEnvironment) {
```

Comment:
might be cleaner to do the exit clause insteadof nesting

if (!this.isBrowserEnv) {
return
}

.... do normal stuff

Reply:
I like this suggestion, will update